jscomp/runtime/caml_bigint.res

```diff
@@ -0,0 +1,37 @@
+/* Copyright (C) 2015-2016 Bloomberg Finance L.P.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * In addition to the permissions granted to you by the LGPL, you may combine
+ * or link a "work that uses the Library" with a publicly distributed version
+ * of this file to produce a combined library or application, then distribute
+ * that combined work under the terms of your choosing, with no requirement
+ * to comply with the obligations normally placed on you by section 4 of the
+ * LGPL version 3 (or the corresponding section of a later version of the LGPL
+ * should you choose to use a later version).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA. */
+
+let div = (x: bigint, y: bigint) =>
+  if y == 0n {
+    raise(Division_by_zero)
+  } else {
+    Caml_bigint_extern.div(x, y)
+  }
+
+let mod_ = (x: bigint, y: bigint) =>
+  if y == 0n {
+    raise(Division_by_zero)
+  } else {
+    Caml_bigint_extern.rem(x, y)
+  }
```

Comment:
I'm not sure this is really necessary, JS bigint already throw its own error (`RangeError` with its message) on div/mod by zero.

So this just duplicates the original behavior, and shadows original error class.

Reply:
This is for catching the res exn as Division_by_zero.
```res
try {
  ...
} catch {
|  Division_by_zero => ...
}
```

Reply:
It's effective for pure ReScript code for sure, but not for external interoperability cases.

The best way I think would be to enable more handling of JS built-in exceptions other than `Js.Exn.Error`, e.g. `Js.Exn.RangeError` or `Js.Exn.TypeError` but the current compiler does not support this.

If we plan to support it in the future, it may be better to postpone this. Detection is possible with `exception Js.Exn.Error`.

Reply:
(It's definitely not a pretty. The only reason I claim this is that bigint is the JS primitive)

Reply:
Yes, it is 🥲 I was trying to match the existing implementation of int.
I don't know if it's possible, but if it becomes possible to bind JS built-in exception with a customizable tagged variant, it would be nice to refactor this.